src/hotspot/share/runtime/arguments.cpp

```diff
@@ -4187,6 +4192,49 @@ jint Arguments::apply_ergo() {
     }
   }
 
+  if (UseCompactObjectHeaders) {
+#if !defined(_LP64) || !(defined(X86) || defined(AARCH64))
+    warning("UseCompactObjectHeaders is not supported with current platform"
+            "; ignoring UseCompactObjectHeaders flag.");
+    FLAG_SET_DEFAULT(UseCompactObjectHeaders, false);
+#else
+    if (FLAG_IS_CMDLINE(UseCompressedClassPointers) && !UseCompressedClassPointers) {
+      warning("UseCompactObjectHeaders is not supported with -XX:-UseCompressedClassPointers"
+              "; ignoring UseCompactObjectHeaders flag.");
+      FLAG_SET_DEFAULT(UseCompactObjectHeaders, false);
+    }
+    if (!(UseG1GC || (UseParallelGC && UseParallelOldGC))) {
+      warning("UseCompactObjectHeaders is not supported with current GC setting"
+              "; ignoring UseCompactObjectHeaders flag.");
+      FLAG_SET_DEFAULT(UseCompactObjectHeaders, false);
+    }
+
+#if INCLUDE_JVMCI
+    if (EnableJVMCI) {
+      warning("UseCompactObjectHeaders is not supported with EnableJVMCI"
+              "; ignoring UseCompactObjectHeaders flag.");
+      FLAG_SET_DEFAULT(UseCompactObjectHeaders, false);
+    }
+#endif
+#if INCLUDE_AOT
+    if (UseAOT) {
+      warning("UseCompactObjectHeaders is not supported with UseAOT"
+              "; ignoring UseCompactObjectHeaders flag.");
+      FLAG_SET_DEFAULT(UseCompactObjectHeaders, false);
+    }
+#endif
+
+    if (UseCompactObjectHeaders) {
```

Comment:
It's duplicated with line 4195

Reply:
UseCompactObjectHeaders might be turned off above